tests/data/service_entry_data.go

```diff
@@ -45,7 +45,7 @@ func CreateEmptyMeshExternalServiceEntry(name, namespace string, hosts []string)
 func AddPortDefinitionToServiceEntry(portDef map[string]interface{}, se kubernetes.IstioObject) kubernetes.IstioObject {
 	if portsSpec, found := se.GetSpec()["ports"]; found {
 		if portsSlice, ok := portsSpec.([]interface{}); ok {
-			portsSlice = append(portsSlice, portDef)
+			_ = append(portsSlice, portDef)
```

Comment:
This doesn't look right to me.  As I understand append the original slice is only re-used if it has capacity.  Otherwise a new array is allocated.  If you don't assign the result of the append it is unsafe unless you are absolutely sure the target slice has capacity.

Reply:
This one took me some time to figure out a fix.
To me, this "smells" like a bug. Using as a reference the original code... What I understand is that if `append()` does not reallocate the slice, everything should work OK. But if `append()` needs to reallocate, then even assigning the result to `portSlice` shouldn't do nothing because assignment is not updating the original handler of the data. Perhaps, the right fix should be:

```go
if portsSlice, ok := portsSpec.([]interface{}); ok {
se.GetSpec()["ports"] = append(portsSlice, portDef)
}
```

I'll let @burmanm to comment, since he is the author.

Reply:
@israel-hdez I think your fix is correct. Looking at this code, it's only for test purpose, and it's currently unused (it always pass through the `else` branch below, skipping that part, so it's unseen. 
I've tested with your suggestion and modifying the test to make it pass through that condition, and it works, so I think it's safe to change as you suggest.

Reply:
ok, changed!